src/lib/coda_graphql/coda_graphql.ml

```diff
@@ -175,6 +184,46 @@ module Types = struct
                 "The transaction has either been snarked, reached finality \
                  through consensus or has been dropped" ]
 
+  let consensus_time : (_, Consensus.Global_slot.t option) typ =
+    let open Consensus.Global_slot in
+    obj "ConsensusTime" ~fields:(fun _ ->
+        [ field "epoch" ~typ:(non_null uint32)
+            ~args:Arg.[]
+            ~resolve:(fun _ global_slot -> epoch global_slot)
+        ; field "slot" ~typ:(non_null uint32)
+            ~args:Arg.[]
+            ~resolve:(fun _ global_slot -> slot global_slot)
+        ; field "startTime" ~typ:(non_null string)
+            ~args:Arg.[]
+            ~resolve:(fun _ global_slot ->
+              Block_time.to_string @@ start_time global_slot )
+        ; field "endTime" ~typ:(non_null string)
+            ~args:Arg.[]
+            ~resolve:(fun _ global_slot ->
+              Block_time.to_string @@ end_time global_slot ) ] )
+
+  let block_proposal :
```

Comment:
I don't remember this field from our conversation -- we had talked about just having a list of known proposal `ConsensusTime`s I think? It seems like the difference between "Propose" and "Propose_now" can easily be checked by seeing if you're meant to propose during the block and "Check_again" would be an empty list.